### PR TITLE
Fix deepspeed ci

### PR DESCRIPTION
**Pull request by SunMarc**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48640) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48640)

# What does this PR do?

This PR fixes the deepseed Ci tests that have been broken for a while due to the dockerfile. Also we fix some training tests.

### CODE REVIEW

**Comment by SunMarc** [comment]
run-slow: trainer

**Comment by ydshieh** [comment]
@askserge Let's make DeepSpeed CI great again 🔥

**Comment by ydshieh** [comment]
Two minor remarks:

**`deepspeed.py` – `max(1, warmup_steps)`:** This is production code (the core integration layer used by `Trainer` whenever DeepSpeed is enabled), so this is a real behavior change for any user training with `warmup_steps=0` (the default) and `scheduler.params.warmup_num_steps: "auto"` in their DeepSpeed config. Concretely, instead of starting at full LR on step 1, they now get one warmup step at LR≈0 first. The change is forced by DeepSpeed ≥0.19.6 rejecting `warmup_num_steps=0`, but I'm not confident whether the new behavior (1-step warmup) is actually correct when training with DeepSpeed — worth a closer look. I'd suggest adding in the code:
```python
# TODO: See PR #48640 — we might need to dive deeper into this behavior change.
```

**`train.py` – `DataCollatorWithPositionIds`:** The flat `arange` is correct for the current setup (Qwen2 + right-padding — Qwen2 does the exact same `arange` internally when `position_ids` is `None`). Just worth keeping in mind that if the script is ever run with a model that derives position IDs from `attention_mask` and with left-padded inputs, the `arange` would assign wrong global positions to the real tokens.

**Review by ydshieh — APPROVED** [review]
LGTM, thanks!
